backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskResource.java

```diff
@@ -171,8 +171,11 @@ public Response copy(Action action) {
 
     @Override
     public Response reduce(Action action) {
-        Disk disk = get();
-        Guid imageId = getDiskImageId(disk.getImageId());
+        Guid imageId = Guid.createGuidFromString(action.getDisk().getImageId());
```

Comment:
doesn't it fail on calls to 'reduce' without specifying a 'disk' in the body? (asking because I'm missing a check for action.isSetDisk here)